src/app/blog/article.service.ts

```diff
@@ -15,12 +16,31 @@ export class ArticleService {
     this.articles = allArticles;
   }
 
-  getAllArticles(): Article[] {
-    return this.articles;
+  getArticles(searchParams: ArticleSearchParams = {}): Observable<Article[]> {
+    let filteredArticles = this.articles;
+    if (searchParams.author) {
+      filteredArticles = filteredArticles.filter(
+        (article) => article.author.id === searchParams.author
+      );
+    }
+
+    if (searchParams.tag) {
+      filteredArticles = filteredArticles.filter((article) =>
+        article.tags.includes(searchParams.tag!)
+      );
+    }
+
+    const articlesSubject = new Subject<Article[]>();
+    setTimeout(() => {
+      articlesSubject.next(filteredArticles);
+      articlesSubject.complete();
+    }, 1000);
+
+    return articlesSubject;
```

Comment:
bad approach. Create class private param with article subject. After that create ``public articles$ = articlesSubject.asObservable()``

Check the training project